app/models/principal.rb

```diff
@@ -73,6 +73,15 @@ class Principal < ApplicationRecord
   }
 
   scope :not_builtin, -> {
+    # TODO: Remove PlaceholderUser from this list. This is a temporary hack that ensures that Placeholders don't
+    # suddenly show up where they are are not supposed to show up. In case you want them to show up use the temporary
+    # scope :not_builtin_but_with_placeholder_users
+    where.not(type: [SystemUser.name, AnonymousUser.name, DeletedUser.name, PlaceholderUser.name])
+  }
+
+  scope :not_builtin_but_with_placeholder_users, -> {
+    # TODO: This is temporary precaution scope to circumvent the hack in the :not_builtin scope. Needs to be
+    # removed before we release this code.
     where.not(type: [SystemUser.name, AnonymousUser.name, DeletedUser.name])
```

Comment:
I know those are temporary. Because of that, can we deprecate them right away so that we do not forget to remove them?
https://github.com/opf/openproject/blob/dev/lib/open_project/deprecation.rb might be handy to do so

Reply:
Tried that. Didn't work straight away. When starting Rails no deprecation warning was shown. Maybe as this is a scope and not an instance method?